cody.el

```diff
@@ -188,11 +190,12 @@ You can call `cody-restart' to force it to re-check the version.")
 
 (defvar cody-mode-map
   (let ((map (make-sparse-keymap)))
-    (define-key map (kbd "C-c c") 'cody-request-completion)
+    (define-key map (kbd "C-c c") 'cody-request-completion) ; prolly a bad default?
```

Comment:
yes this is a bad default. `C-c letter` is reserved for users keymapping. Given we are a minor mode there is guidance on what we couuld do. https://www.gnu.org/software/emacs/manual/html_node/elisp/Key-Binding-Conventions.html

For me this overwrites my org-mode capture binding, which was quite frustrating when I first tried this out.

Reply:
TIL.

That page simultaneously: 1. makes perfect sense, 2. triggers a "STOP BINDING KEYS IN EMACS/THEY HAVE PLAYED US FOR ABSOLUTE FOOLS" vibe.

Reply:
Can anyone suggest some good default bindings?

Reply:
TL;DR: How about `C-c /`?

I looked around for existing keybindings in similar packages, but there was nothing obvious.

In [copilot.el](https://github.com/zerolfx/copilot.el), there is no keybinding specified. You can either rely on the automatic idle timer completion, or otherwise the docs suggest that you create your own keybinding for the `copilot-complete` command (but they don't specify a particular key).

In [codeium.el](https://github.com/Exafunction/codeium.el), no keybindings are needed because it simply provides a backend to the built-in `completion-at-point` framework.  The package provides a function that you add to `completion-at-point-functions`, then you can use the built-in `M-/` (`completion-at-point`) keybinding and/or use completion frontend packages such as `corfu` or `company-mode`.

In the Emacs documentation, there is a [subsection for minor modes](https://www.gnu.org/software/emacs/manual/html_node/elisp/Keymaps-and-Minor-Modes.html).  It has much the same guidance as the section on major modes linked above:

> Minor modes may bind commands to key sequences consisting of C-c followed by a punctuation character. However, sequences consisting of C-c followed by one of {}<>:;, or a control character or digit, are reserved for major modes. Also, C-c letter is reserved for users. See Key Binding Conventions.

This led me to suggest `C-c /`.  It evokes the default `M-/` keybinding and it's legal for minor modes to use.  It's not bound to anything in my Emacs and I couldn't find anything obvious by searching (but admittedly it's a hard one to search for). Thoughts?

As a related point, it would be fantastic if cody.el could provide a `completion-at-point-functions` endpoint!  I'm working on a PR for corfu that makes it play better with multi-line completions.  It has been working really well with `codeium.el` completions, but I'd much prefer Cody.

Reply:
@jrblevin I love C-c / -- this is an amazing suggestion. It's reserved for minor modes; it doesn't seem to be used by anything popular; it's easy to type. Great stuff.

I do plan to integrate with completion-at-point and maybe company-mode.